packages/javascript/bh-shared-ui/src/components/AssetGroupFilters/AssetGroupFilters.test.tsx

```diff
@@ -0,0 +1,171 @@
+        await waitFor(() => expect(expandedSection.classList.contains('MuiCollapse-entered')).toBeTruthy());
+    });
+
+    it('indicates that filters are active', async () => {
+        const { screen } = await setup({ filterParams, availableNodeKinds });
+
+        const filtersContainer = screen.getByTestId('asset-group-filters-container');
+
+        expect(filtersContainer.className.includes('activeFilters')).toBeTruthy();
+    });
+
+    it('indicates that filters are inactive', async () => {
+        const { screen } = await setup();
+
+        const filtersContainer = screen.getByTestId('asset-group-filters-container');
+
+        expect(filtersContainer.className.includes('activeFilters')).toBeFalsy();
+    });
+
+    describe('Node Type dropdown filter', () => {
+        it('displays the value from filterParams.node_type', async () => {
+            const { screen } = await setup({ filterParams, availableNodeKinds });
+            const nodeTypeFilter = screen.getByTestId('asset-groups-node-type-filter');
+            const nodeTypeFilterValue = nodeTypeFilter.firstChild?.nextSibling;
+
+            expect(nodeTypeFilter.textContent).toContain('Domain');
+            expect((nodeTypeFilterValue as HTMLInputElement)?.value).toBe('eq:Domain');
+        });
+
+        it('lists all available node kinds as options to filter by', async () => {
+            const { screen, user } = await setup({ availableNodeKinds });
+
+            await user.click(screen.getByTestId('display-filters-button'));
+            await user.click(screen.getByLabelText('Node Type'));
+
+            const nodeKindList = await screen.findAllByRole('option');
+
+            expect(nodeKindList).toHaveLength(availableNodeKinds.length + 1); // +1 for the default empty value
+
+            for (const nodeKind of availableNodeKinds) {
+                expect(screen.getByText(nodeKind)).toBeInTheDocument();
+            }
+        });
+
+        it('calls handleFilterChange when a node type is selected', async () => {
+            const { screen, user, handleFilterChange } = await setup({ availableNodeKinds });
+
+            await user.click(screen.getByTestId('display-filters-button'));
+            await user.click(screen.getByLabelText('Node Type'));
+            await user.click(screen.getByText(availableNodeKinds[0]));
+
+            expect(handleFilterChange).toBeCalledTimes(1);
+            expect(handleFilterChange).toHaveBeenCalledWith('primary_kind', `eq:${availableNodeKinds[0]}`);
+        });
+    });
+
+    describe('Custom Member checkbox filter', () => {
+        it("displays the checkbox as checked if the filter params value is 'true'", async () => {
+            const { screen } = await setup({ filterParams, availableNodeKinds });
+            const checkbox = screen.getByTestId('asset-groups-custom-member-filter');
+
+            expect((checkbox.firstChild as HTMLInputElement)?.checked).toBe(true);
```

Comment:
Should we explicitly set the filterParams value to true here instead of trusting that the mock factory provides this value?

Reply:
Yeah great suggestion! And it's more explicit